tfs/src/main/java/hudson/plugins/tfs/rm/ReleaseManagementCI.java

```diff
@@ -163,6 +241,36 @@ void CreateRelease(
             JSONObject object = new JSONObject(response);
             listener.getLogger().printf("Release Name: %s%n", object.getString("name"));
             listener.getLogger().printf("Release id: %s%n", object.getString("id"));
+
+            if (StringUtils.isNotBlank(this.destinationPath)) {
+                listener.getLogger().printf("Waiting for release to be finished...%n");
+                while (true) {
+                    String status = releaseManagementHttpClient.GetReleaseStatus(this.projectName, object.getString("id"));
+                    if (status.equals("finished")) {
```

Comment:
what if someone cancelled the release or deleted it. This will forever check the status. We should handle other status as well.

Also, what if there is a post approval which has not been approved for days. This will continue to check for that many days. We should also have a timeout to stop checking. 

For one of the above reason if the release is not succeeded and we stopped checking for status and no logs uploaded to jenkins workspace, in that case whenever the jenkins build page is opened we can check once again to see the current status of that release and update the jenkins workspace. is this achievable?

Reply:
I will check more status and add a time watcher. Do you know where to get documents about the all types of status. I can't find it.
For the last suggestion, I think it's hard to achieve. Because the environment may be changed a lot after build. The workspace may no longer exists after build (User may use dynamic slave and once the build is over, the slave will be deleted) or the configuration has changed. So deal with the old builds may cause unexpected issues. Thus, I think it's make sense to skip the unachievable logs.

Reply:
It is there in Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.EnvironmentStatus
Possible status are,
Undefined, NotStarted, InProgress, Succeeded, Canceled, Rejected, Queued, Scheduled, PartiallySucceeded